Source/Scene/Scene.js

```diff
@@ -3886,12 +4064,92 @@ define([
         }
         //>>includeEnd('debug');
         var ray = getRayForClampToHeight(this, cartesian);
-        var pickResult = pickFromRay(this, ray, objectsToExclude, true);
+        var pickResult = pickFromRay(this, ray, objectsToExclude, true, false);
         if (defined(pickResult)) {
             return Cartesian3.clone(pickResult.position, result);
         }
     };
 
+    /**
+     * Initiates an asynchronous {@link Scene#sampleHeight} query for an array of {@link Cartographic} positions
+     * using the maximum level of detail for 3D Tilesets in the scene. Returns a promise that is resolved when
+     * the query completes. Each point height is modified in place. If a height cannot be determined because no
+     * geometry can be sampled at that location, or another error occurs, the height is set to undefined.
+     *
+     * @param {Cartographic[]} positions The cartographic positions to update with sampled heights.
+     * @param {Object[]} [objectsToExclude] A list of primitives, entities, or features to not sample height from.
+     * @returns {Promise.<Number[]>} A promise that resolves to the provided list of positions when the query has completed.
+     *
+     * @see Scene#sampleHeight
+     *
+     * @exception {DeveloperError} sampleHeightMostDetailed is only supported in 3D mode.
+     * @exception {DeveloperError} sampleHeightMostDetailed requires depth texture support. Check sampleHeightSupported.
+     */
+    Scene.prototype.sampleHeightMostDetailed = function(positions, objectsToExclude) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('positions', positions);
+        if (this._mode !== SceneMode.SCENE3D) {
+            throw new DeveloperError('sampleHeightMostDetailed is only supported in 3D mode.');
+        }
+        if (!this.sampleHeightSupported) {
+            throw new DeveloperError('sampleHeightMostDetailed requires depth texture support. Check sampleHeightSupported.');
+        }
+        //>>includeEnd('debug');
+        objectsToExclude = defined(objectsToExclude) ? objectsToExclude.slice() : objectsToExclude;
+        var i;
```

Comment:
Why reuse `i` in two loops?  Just use `i` and `j`.  Minimize the span of variable use even if it is just "optics" in this version of JS.

Reply:
It's also being accessed via closure below which is slower.